System/Atom.cpp

```diff
@@ -0,0 +1,1060 @@
+#include "Atom.h"
+#include "SLTerrain.h"
+#include "MovableObject.h"
+#include "MOSRotating.h"
+#include "PresetMan.h"
+#include "Actor.h"
+
+namespace RTE {
+
+	const std::string Atom::c_ClassName = "Atom";
+	std::vector<void *> Atom::s_AllocatedPool;
+	int Atom::s_PoolAllocBlockCount = 200;
+	int Atom::s_InstancesInUse = 0;
+
+	// This forms a circle around the Atom's offset center, to check for mask color pixels in order to determine the normal at the Atom's position.
+	const int Atom::s_NormalChecks[c_NormalCheckCount][2] = { {0, -3}, {1, -3}, {2, -2}, {3, -1}, {3, 0}, {3, 1}, {2, 2}, {1, 3}, {0, 3}, {-1, 3}, {-2, 2}, {-3, 1}, {-3, 0}, {-3, -1}, {-2, -2}, {-1, -3} };
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void Atom::Clear() {
+		m_Offset.Reset();
+		m_OriginalOffset.Reset();
+		m_Normal.Reset();
+		m_Material = g_SceneMan.GetMaterialFromID(g_MaterialAir);
+		m_SubgroupID = 0;
+		m_MOHitsDisabled = false;
+		m_TerrainHitsDisabled = false;
+		m_OwnerMO = 0;
+		m_IgnoreMOID = g_NoMOID;
+		m_IgnoreMOIDs.clear();
+		m_MOIDHit = g_NoMOID;
+		m_TerrainMatHit = g_MaterialAir;
+		m_LastHit.Reset();
+		/*
+		m_HitVel.Reset();
+		m_HitRadius.Reset();
+		m_HitImpulse.Reset();
+		*/
+		m_TrailColor.Reset();
+		m_TrailLength = 0;
+		m_NumPenetrations = 0;
+		m_ChangedDir = true;
+		m_ResultWrapped = false;
+		m_PrevError = 0;
+		m_StepRatio = 1.0F;
+		m_SegProgress = 0.0F;
+
+		m_IgnoreMOIDsByGroup = 0;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	int Atom::Create(const Vector &offset, Material const *material, MovableObject *owner, Color trailColor, int trailLength) {
+		m_Offset = m_OriginalOffset = offset;
+		// Use the offset as normal for now
+		m_Normal = m_Offset;
+		m_Normal.Normalize();
+		m_Material = material;
+		m_OwnerMO = owner;
+		m_TrailColor = trailColor;
+		m_TrailLength = trailLength;
+
+		return 0;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	int Atom::Create(const Atom &reference) {
+		//Entity::Create(reference);
+
+		m_Offset = reference.m_Offset;
+		m_OriginalOffset = reference.m_OriginalOffset;
+		m_Normal = reference.m_Normal;
+		m_Material = reference.m_Material;
+		m_SubgroupID = reference.m_SubgroupID;
+		// Needs to be set manually by the new MO owner.
```

Comment:
They're pretty informative.

Reply:
Yeah, but I wonder whether it's clear or not without them, just by the code. They're not bad though, not as useless as '//Setting owner to 0' or something that tells you literally nothing the code doesn't. I guess leave em be if you like em.

Reply:
I'm gonna leave them be because they'll save us the "Why are these two not being copied over like the rest?" and the research that follows when we look at it later:tm:.

Reply:
Hahaha sure, fair point :D